### PR TITLE
Delta Lake Diff: Change default plugin loading

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
## Change Description

First, look up the default directory for an executable named `delta` and register it if it's there.
Then test the different diff plugin configurations available and run over the previous registration if needed.

Why not the other way around?
Because it's easier and doesn't cost too much...

### CODE REVIEW

**Review by arielshaqed — APPROVED** [review]
# Nice!

File: pkg/plugins/diff/service.go [hunk 1]
```diff
@@ -142,14 +146,20 @@ func NewService(diffProps map[string]config.DiffProps, pluginProps map[string]co
...
-		if props, ok := pluginProps[pluginName]; ok {
-			pluginPath = props.Path
+		if props, ok := pluginProps.Properties[pluginName]; ok {
+			pp, err := homedir.Expand(props.Path)
+			if err != nil {
+				logging.Default().Errorf("failed to register a plugin for an invalid path: '%s'", props.Path)
+				continue
+			}
+			pluginPath = pp
 			if props.Version != nil {
```
Comment (nopcoder) [inline pkg/plugins/diff/service.go:163]: I think we can remove the Version pointer from the configuration as we fallback to version 1 in the code - we can document that 0 is the default.

File: pkg/plugins/diff/service.go [hunk 2]
```diff
@@ -166,3 +176,25 @@ func registerPlugins(service *Service, diffProps map[string]config.DiffProps, pl
 		logging.Default().Infof("successfully registered a plugin for diff type: '%s'", n)
 	}
 }
+
+func registerDefaultPlugins(service *Service, pluginsPath string) {
+	diffPluginsDir := diffPluginsDefaultPath(pluginsPath)
+	deltaPath := filepath.Join(diffPluginsDir, "delta")
+	_, err := os.Stat(deltaPath)
+
+	if err != nil {
```
Comment (nopcoder) [inline pkg/plugins/diff/service.go:185]:
```suggestion
	_, err := os.Stat(deltaPath)
	if err != nil {
```

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/config/config.go [hunk 3]
```diff
@@ -326,8 +332,8 @@ type Config struct {
 			Code string `mapstructure:"code"`
 		} `mapstructure:"snippets"`
 	} `mapstructure:"ui"`
-	Diff    map[string]DiffProps   `mapstructure:"diff"`
-	Plugins map[string]PluginProps `mapstructure:"plugins"`
+	Diff    map[string]DiffProps `mapstructure:"diff"`
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply pkg/config/config.go:335]: Created an [issue](https://github.com/treeverse/lakeFS/issues/5518) for that and will be handled in the following iteration.

**Pull request merged**